File: 品味Java的21種設計模式/dp_java/abstractfactory/example3/IntelCPU.java
package cn.javass.dp.abstractfactory.example3;
/**
 *Intel的CPU实现
 */
public class IntelCPU implements CPUApi {
    /**
     * CPU的针脚数目
     */
    private int pins = 0;
    /**
     * 构造方法，传入CPU的针脚数目
     * @param pins CPU的针脚数目
     */
    public IntelCPU(int pins) {
        this.pins = pins;
    }

    public void calculate() {
        System.out.println("now in Intel CPU,pins=" + pins);
    }
}

File: 品味Java的21種設計模式/dp_cpp/abstractfactory/example3/IntelCPU.cpp
#include "IntelCPU.h"
#include <iostream>

namespace cn {
namespace javass {
namespace dp {
namespace abstractfactory {
namespace example3 {

IntelCPU::IntelCPU(int pins) {
    this->pins = pins;
}

void IntelCPU::calculate() {
    std::cout << "now in Intel CPU,pins=" << pins << std::endl;
}
}
}
}
}
}
